Mathlib/Tactic/Linter/CasesLinter.lean

```diff
@@ -0,0 +1,54 @@
+/-
+Copyright (c) 2025 Jeremy Tan. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Jeremy Tan, Damiano Testa
+-/
+
+import Lean.Elab.Command
+import Mathlib.Tactic.Cases
+
+/-!
+# The "cases" linter
+
+The "cases" linter flags uses of the `cases'` tactic, which is a backward-compatible version of
+Lean 3's `cases` tactic. Unlike `obtain`, `rcases` and Lean 4's `cases`, variables introduced by
+`cases'` are not required to be separated by case, which hinders readability.
+
+This linter is an incentive to discourage uses of `cases'`, without being a ban.
+-/
+
+open Lean Elab
+
+namespace Mathlib.Linter
```

Comment:
I think this linter should be in the `Mathlib.Linter.Style` namespace, and the option be called `linter.style.cases` (or even `linter.style.cases'`). I know you're just going off the refine' linter; that linter should also be moved.